src/chat-handler.ts

```diff
@@ -87,39 +87,44 @@ export class ChatHandler extends ChatModel {
       })
     );
 
-    this.updateWriters([{ username: 'AI' }]);
-    return this._aiProvider.chatModel
-      .invoke(messages)
-      .then(response => {
-        // Some providers may use text-completion models (for example ChromeAI),
-        // for which the response is a string.
-        const content = response.content ?? response;
-        const botMsg: IChatMessage = {
-          id: UUID.uuid4(),
-          body: content.toString(),
-          sender: { username: 'AI' },
-          time: Date.now(),
-          type: 'msg'
-        };
+    const sender = { username: 'AI' };
+    this.updateWriters([sender]);
+
+    // create an empty message to be filled by the AI provider
+    const botMsg: IChatMessage = {
+      id: UUID.uuid4(),
+      body: '',
+      sender,
+      time: Date.now(),
+      type: 'msg'
+    };
+
+    let content = '';
+
+    try {
+      for await (const chunk of await this._aiProvider.chatModel.stream(
+        messages
+      )) {
+        content += chunk.content;
```

Comment:
Wondering if the stream are always additional or could try to modify previous part of the message.
Don't know if there is a setting for that, and if it worth looking for that feature in future.

Reply:
Guessing they should be additional as otherwise it would be difficult to tell if that's the case or not?

Reply:
Or you mean if some providers would return the complete updated message each time?

For example:

- He
- Hello
- Hello,
- Hello, world

Instead of:

- He
- llo
- ,
- world

The second one is more efficient in terms of bandwidth and data being transferred, so there may not be many advantages for the first approach?

Reply:
In this case I agree that additional is the best approach.

> Guessing they should be additional as otherwise it would be difficult to tell if that's the case or not?

Indeed, it must always act in the same way.

To explain my question, it looks like it's writing word by word, like a human. But sometime you realise in the middle of the sentence that you could have wrote it differently for more consistency, and want to reformat the whole sentence. I was wondering if some model could act similarly to human. But as you said, it would be difficult to know with this API.